### PR TITLE
AIP#31 impl changes for all crates

### LINKED ISSUE #312: Remove struct TargetingTag
Since the introduction of [AIP#31](https://github.com/AdExNetwork/aips/issues/31) we no longer need it, so find and remove all places and mentions of it.
The `TargetingTag` is contained in it's own module so the whole module should also be removed.

**Note:** It requires changes to `adview-manager` crate as well

### PR DESCRIPTION
**Pull request by elpiel**
Resolves #312 
Partially impls #323 #322 #319 

Related #340 

### adview-manager:
- [x] impl changes on `adview-manager` crate
- *Other tasks are moved into #322!*

#### ~Event aggregator:~
- [ ] ~Check impl of Event Aggregator~
- [ ] ~Test changes~
#### Event Reducer
- [ ] ~Check impl of Event Reducer~
- [ ] ~Test changes~
#### Analytic Recorder:
- [ ] ~Check impl of Analytics Recorder~
- [ ] ~Tests changes~

#### get_payout:
- [x] impl JS tests for `get_payout()` https://github.com/AdExNetwork/adex-validator/blob/476aecdd64c7e1f535ff7991aeae9c18c045b6db/test/index.js#L359-L364
- [ ] ~More Tests for `get_payout()`~ Moved to #323 

## Partially resolves:
- #323 impl AIP#31 changes to Payout  
- #322 adview-manager: AIP#31 Targeting changes
- #319 EventAggregator changes

### CODE REVIEW

**Comment by elpiel** [comment]
Code todos that need address:
- [ ] https://github.com/AdExNetwork/adex-adview-manager/blob/master/src/main.ts#L16-L20
- [ ] Default empty string for `Input::navigator_language` See: https://github.com/AdExNetwork/adex-validator-stack-rust/blob/7fd8a4b9c39ba9cbb57367c7a90f07823af7643d/adview-manager/src/lib.rs#L329-L330
- [ ] Double-check if this `.expect()` is reasonable. See:
https://github.com/AdExNetwork/adex-validator-stack-rust/blob/7fd8a4b9c39ba9cbb57367c7a90f07823af7643d/adview-manager/src/lib.rs#L358-L368

- [ ] Fix / Check `.unwrap()` and what should happen if we don't find the `ValidatorDesc`. See:
https://github.com/AdExNetwork/adex-validator-stack-rust/blob/7fd8a4b9c39ba9cbb57367c7a90f07823af7643d/adview-manager/src/lib.rs#L548-L558

**Review by elpiel — COMMENTED** [review]
Todos mentioned in the comment:
https://github.com/AdExNetwork/adex-validator-stack-rust/pull/313#issuecomment-693445256

**Comment by elpiel** [comment]
ping @simzzz @samparsky 
Please review the PR and provide me with feedback on the remaining questions / tasks.
**NB:** Some tasks were moved to the issues to unblock this PR

**Review by samparsky — APPROVED** [review]

File: adview-manager/src/lib.rs [hunk 1]
```diff
@@ -1,203 +1,96 @@
...
+lazy_static! {
+// Impression "stickiness" time: see https://github.com/AdExNetwork/adex-adview-manager/issues/65
+// 4 minutes allows ~4 campaigns to rotate, considering a default frequency cap of 15 minutes
+    pub static ref IMPRESSION_STICKINESS_TIME: chrono::Duration = chrono::Duration::milliseconds(240000);
+}
+
 #[derive(Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct AdViewManagerOptions {
+// TODO: Add Default Ops somehow?
```
[the remark this answers is not in the document]
Comment (elpiel) [reply adview-manager/src/lib.rs:45]: Moved to #322

File: adview-manager/src/lib.rs [hunk 2]
```diff
@@ -217,106 +118,499 @@ fn video_html(
...
+            .units_with_price
+            .iter()
+            .find_map(|u| {
+                if u.unit.id == sticky_entry.unit_id {
+                    Some(u.unit.clone())
+                } else {
+                    None
+                }
+            })
+            .expect("Something went terribly wrong. Data is corrupted! There should be an AdUnit");
```
[the remark this answers is not in the document]
Comment (elpiel) [reply adview-manager/src/lib.rs:368]: Moved to #322

File: adview-manager/src/lib.rs [hunk 3]
```diff
@@ -217,106 +118,499 @@ fn video_html(
...
+            let validators = campaigns
+                .iter()
+                .find_map(|campaign| {
+                    if &campaign.channel.id == campaign_id {
+                        Some(&campaign.channel.spec.validators)
+                    } else {
+                        None
+                    }
+                })
+                // TODO: Check what should happen here if we don't find the Validator
```
[the remark this answers is not in the document]
Comment (elpiel) [reply adview-manager/src/lib.rs:557]: Moved to #322

**Pull request merged**